### PR TITLE
less aggressive sanity check for reward functions of hindsight and en…

### LINKED ISSUE #1118: Hindsight experience transformer warns incorrectly
Hi,

I'm using HER to train the `fetchpush` task.
I get a very long warning message at every training step.
The message is created here:
https://github.com/HorizonRobotics/alf/blob/ee1e020ee732468d2f94629b0ad29feb68090075/alf/algorithms/data_transformer.py#L823-L835
If I understand this correctly, the condition checks if not all new rewards are the same as the old rewards.
Since the task of HER is to create new rewards based on new goals, this check should be exactly the opposite
```
if torch.allclose(relabeled_rewards[non_her_indices],
                              result.reward[non_her_indices]):
 ** WARN **
```
instead of 
```
if not torch.allclose(relabeled_rewards[non_her_indices],
                              result.reward[non_her_indices]):
 ** WARN **
```

The bug should be reproducible on the mujoco docker using [this gin file](https://github.com/HorizonRobotics/alf/blob/pytorch/alf/examples/her_fetchpush.

Did I get this correctly?

Best,

Jakob

### PR DESCRIPTION
**Pull request by le-horizon**
Fixes #1118
This change doesn't impact performance, only logging warning.

### CODE REVIEW

File: alf/algorithms/data_transformer_test.py [hunk 1]
```diff
@@ -226,8 +226,7 @@ def test_compute_her_future_step_distance(self, end_prob):
             data_spec=self.data_spec,
             num_environments=num_envs,
             max_length=max_length,
-            keep_episodic_info=True,
-            step_type_field="t")
```
Comment (hnyu) [inline alf/algorithms/data_transformer_test.py:230]: Why did you have this `step_type_field` in the first place?

Comment (le-horizon) [reply alf/algorithms/data_transformer_test.py:230]: So the test can have a different field name for step_type, which is no longer used due to this PR.
Just removed it.

**Review by hnyu — APPROVED** [review]
